csrc/scheduler/normalization_utils.cpp

```diff
@@ -1235,14 +1236,32 @@ void movePersistentBufferToSmem(
       // path of gmem -> smem to reduce temporary register usage. Otherwise, the
       // data path from gmem to shared memory (smem) follows this sequence: gmem
       // -> L1 cache -> register -> smem.
+      // Only supported after device 8.0 and requires vectorized load.
       int hw_major = at::cuda::getCurrentDeviceProperties()->major;
-      if (is_cached_input && hw_major >= 8) {
+      if (rparams->vectorize_inner_reduction && is_cached_input &&
+          hw_major >= 8) {
         tv->definition()->as<LoadStoreOp>()->setOpType(
             LoadStoreOpType::CpAsync);
         tv->definition()->as<LoadStoreOp>()->setCacheOp(CacheOp::Unspecified);
       }
+      // T4_s_float = CpAsync(T0_g_float)
```

Comment:
Is this line relevant?

Reply:
removed, was used to help remembering which tensor I was working on.